### DISCUSSION TITLE
Cancel Booking Link

### DISCUSSION THREAD

**Discussion by carlbannister** [liedekef/events-made-easy#281] [Q&A] [answered]
Hi, I have used the cancel booking link in the pending email that is sent out. It works as expected and cancels the unpaid booking. I have just had an instance where the booker has used the link after the booking has been approved and paid. I was expecting that the booking would be cancelled and refunded via the payment gateway (stripe) but it wasn't, only the booking was deleted without refund. Is this expected behaviour? if so any suggestions on how to make it work as i may be missing something.

Thanks

Carl

**Comment by liedekef** [comment, marked as answer]
When the user cancels, no refund happens. It is in fact the same in the backend (refund is by default not done, you need to select it).
Currently this is considered to be dangerous (certainly if the decision to refund is "made" by the booker ...).

**Comment by carlbannister** [comment]
OK that makes sense, hadn't thought of it like that before.

Thanks

**Comment by carlbannister** [comment]
Is there a way to use the cancel booking link only if the booking is
pending. If it is paid the link won't work

**Reply by liedekef** [reply]
I don't understand what you're saying here. Do you mean the cancel is not working if it is paid? Or do you just want an option to cancel if the booking is not yet paid for yet?

**Reply by carlbannister** [reply]
Just the option to cancel the booking if it is not paid yet

**Reply by liedekef** [reply]
While that is possible, it requires new code (option per event). While I have my doubts about the actual use, I'll only implement it after a donation (donations are currently down to 0 in the last 2,5 months, so new features need to contribute to further development :-) ).
